lib/modules/platform/gitlab/readme.md

```diff
@@ -2,39 +2,45 @@
 
 ## Authentication
 
-First, [create a Personal Access Token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html) for the bot account.
+You can authenticate Renovate to GitLab, with a Personal Access Token, or Group Access Token.
 
-If you are using a Group access token, the token must have Developer role or higher permissions in order to create issues and merge requests.
-The token must have Maintainer permissions in order to perform Automerge.
+To start, create either:
 
-For real runs, give the PAT these scopes:
+- a [Personal Access Token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html) for the bot account
+- or a [Group Access Token](https://docs.gitlab.com/ee/user/group/settings/group_access_tokens.html#bot-users-for-groups) for the bot account
+
+The bot account must have at least the Developer role in order to create issues and merge requests.
+The bot account must have the Maintainer role in order to perform automerge.
```

Comment:
```suggestion
The bot account must have the Developer and Maintainer roles to work.

| Permission | Required | Needed to                        |
| ---------- | -------- | -------------------------------- |
| Developer  | Yes      | Create issues and Merge Requests |
| Maintainer | Yes      | Perform automerge                |
```

How about putting the info in a table? Optionally we can hammer home the need for the permissions by putting a Required=yes column in the table.

It's not clear to me if Renovate needs the _full permissions_ of the Developer/Maintainer role, or only _parts_ of it.

Reply:
Hm. I am not sure about this.

An account can only have one role. The role has more permissions than are actually needed. Developer is required to create merge requests and branches, and push to them (see the table here: https://docs.gitlab.com/ee/user/permissions.html#project-members-permissions; for example: "Merge Requests: Create" and "Repository: Create new branches").

The "automerge needs Maintainer" actually depends on how the default branch is setup. It is assumed here that it is a protected branch I think. If I am not mistaken, the default permissions for a protected branch are that to push to it directly you need the Maintainer role. I assume that automerge merges via `git` (not via merge through the GitLab API)? There is a separate rule on protected branches for "allow to merge".

How about changing the second sentence to clarify that if automerge is used the role has to be in line with the protected branch rules (since it is project specific)?

Reply:
I confirm that the Maintainer role is required if the branch protection rules say so. I have the same token without Maintainer role that successfully automerge in some projects but not in others because all my projects don't have the same branch protection rules.

> I assume that automerge merges via git (not via merge through the GitLab API)? 
I don't think so. Automerge enables a "flag" on the MR. It's Gitlab that does the automerge once MR checks are successful.

Reply:
The [Renovate docs, `platformAutomerge`](https://docs.renovatebot.com/configuration-options/#platformautomerge) basically says:

- `platformAutomerge` has a `default` value of `true`
- `platformAutomerge` speeds up merging as it uses GitLab's native automerge functionality
- On GitLab Renovate will re-enable the PR for platform-native automerging after the PR is rebased
- `platformAutomerge` will configure PRs to be merged after the branch policies are met. Works on GitLab.
- Read the platform specific docs for version requirements

So I think this means that Renovate merges "however GitLab merges things".

Reply:
Thanks to you both for the details. I replaced the second sentence with a clarification about the protect branch rules to make it clear that it depends on the "Allowed to merge" setting on the protected branch.